File: tools/sync_vault.js
import fs from 'fs';
import path from 'path';
import dotenv from 'dotenv';
import axios from 'axios';

dotenv.config();

const API_KEY = process.env.GDRIVE_API_KEY;
const FOLDER_ID = process.env.GDRIVE_FOLDER_ID;
const OUTPUT_FILE = path.join(process.cwd(), 'public', 'documents.json');

async function fetchGDriveFiles() {
    console.log(`🚀 Fetching files from GDrive folder: ${FOLDER_ID}`);

    const url = 'https://www.googleapis.com/drive/v3/files';
    try {
        const response = await axios.get(url, {
            params: {
                q: `'${FOLDER_ID}' in parents and trashed = false`,
                fields: 'files(id, name, mimeType, size, webViewLink, thumbnailLink, createdTime)',
                key: API_KEY,
                pageSize: 1000,
            },
        });

        const files = response.data.files || [];

        const formattedItems = files.map((file) => {
            let category = 'Other';
            const nameLower = file.name.toLowerCase();
            if (
                nameLower.includes('id') ||
                nameLower.includes('card') ||
                nameLower.includes('passport') ||
                nameLower.includes('aadhar')
            ) {
                category = 'Identity';
            } else if (
                nameLower.includes('tax') ||
                nameLower.includes('invoice') ||
                nameLower.includes('bill')
            ) {
                category = 'Finance';
            } else if (
                nameLower.includes('medical') ||
                nameLower.includes('report')
            ) {
                category = 'Health';
            } else if (
                nameLower.includes('education') ||
                nameLower.includes('result') ||
                nameLower.includes('certificate')
            ) {
                category = 'Education';
            }

            return {
                id: file.id,
                name: file.name,
                mimeType: file.mimeType,
                size: parseInt(file.size || '0'),
                webViewLink: file.webViewLink,
                thumbnailLink: file.thumbnailLink || '',
                category,
                createdTime: file.createdTime,
            };
        });

        const manifest = {
            lastUpdated: new Date().toISOString(),
            version: '1.0.0',
            items: formattedItems,
        };

        if (!fs.existsSync(path.dirname(OUTPUT_FILE))) {
            fs.mkdirSync(path.dirname(OUTPUT_FILE), { recursive: true });
        }

        fs.writeFileSync(OUTPUT_FILE, JSON.stringify(manifest, null, 2));

        console.log(`✅ Successfully synced ${formattedItems.length} documents to ${OUTPUT_FILE}`);
    } catch (error) {
        console.error(`❌ Error fetching files: ${error.response?.data?.error?.message || error.message}`);
    }
}

fetchGDriveFiles();

File: tools/sync_vault.py
import os
import json
import requests
from dotenv import load_dotenv

load_dotenv()

API_KEY = os.getenv("GDRIVE_API_KEY")
FOLDER_ID = os.getenv("GDRIVE_FOLDER_ID")
OUTPUT_FILE = "public/documents.json"

def fetch_gdrive_files():
    print(f"🚀 Fetching files from GDrive folder: {FOLDER_ID}")
    
    url = "https://www.googleapis.com/drive/v3/files"
    params = {
        "q": f"'{FOLDER_ID}' in parents and trashed = false",
        "fields": "files(id, name, mimeType, size, webViewLink, thumbnailLink, createdTime)",
        "key": API_KEY,
        "pageSize": 1000
    }
    
    response = requests.get(url, params=params)
    
    if response.status_code != 200:
        print(f"❌ Error fetching files: {response.text}")
        return
    
    files = response.json().get("files", [])
    
    formatted_items = []
    for file in files:
        # Simple category logic based on name or mimeType
        category = "Other"
        name_lower = file['name'].lower()
        if "id" in name_lower or "card" in name_lower or "passport" in name_lower or "aadhar" in name_lower:
            category = "Identity"
        elif "tax" in name_lower or "invoice" in name_lower or "bill" in name_lower:
            category = "Finance"
        elif "medical" in name_lower or "report" in name_lower:
            category = "Health"
        elif "education" in name_lower or "result" in name_lower or "certificate" in name_lower:
            category = "Education"

        formatted_items.append({
            "id": file['id'],
            "name": file['name'],
            "mimeType": file['mimeType'],
            "size": int(file.get('size', 0)),
            "webViewLink": file['webViewLink'],
            "thumbnailLink": file.get('thumbnailLink', ""),
            "category": category,
            "createdTime": file['createdTime']
        })
    
    manifest = {
        "lastUpdated": file.get('createdTime') if files else "N/A", # Placeholder for real timestamp
        "version": "1.0.0",
        "items": formatted_items
    }
    
    os.makedirs(os.path.dirname(OUTPUT_FILE), exist_ok=True)
    with open(OUTPUT_FILE, "w") as f:
        json.dump(manifest, f, indent=2)
    
    print(f"✅ Successfully synced {len(formatted_items)} documents to {OUTPUT_FILE}")

if __name__ == "__main__":
    fetch_gdrive_files()
